### PR TITLE
fix(commands): drop stale review comments before posting

### PR DESCRIPTION
**Pull request by rios0rios0**
## Summary

- Re-fetches `provider.GetPullRequestFiles` immediately before posting and drops AI findings whose path is no longer in the latest iteration.
- Logs the dropped count + unique paths at `Warn` (capped to 8 + `(+N more)` sentinel).
- Best-effort: if the re-fetch itself fails, falls back to posting all comments — bot is never worse than today's baseline.
- 14 new unit tests pin every classification edge case via three exported helpers (`filterStaleComments`, `summarizeStaleFilePaths`, `normalizeFilePath`).

## Why

The user reported every bot comment on `internal-app/internal-integrator#NNNN` rendering with ADO's *"this file no longer exists in the latest pull request changes"* warning. Root cause: the diff is fetched at the start of `Execute`; if the user pushes another commit (rebase / squash / follow-up) before `postComments` runs, the AI's findings reference paths the latest iteration no longer touches.

This is a class of staleness that can't be fully solved without ADO iteration context (line drift in still-present files would also need it), but file-level removal — the most visible case — is fixable today with a simple membership check.

## Test plan

- [x] `go test -tags unit ./...` (all green; 14 new test rows in `TestFilterStaleComments` / `TestSummarizeStaleFilePaths` / `TestNormalizeFilePath`)
- [x] `make lint` (0 issues)
- [ ] Merge → `delivery-docker` rebuilds `latest`
- [ ] `kubectl rollout restart deployment/code-guru -n code-guru`
- [ ] Force-push a tracked PR (e.g. squash commits) and verify the bot posts no stale-anchored comments after the squash

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/domain/commands/review_command.go [hunk 1]
```diff
@@ -288,6 +291,104 @@ func (c *ReviewCommand) postComments(
...
+	comments []entities.ReviewComment,
+	livePaths map[string]struct{},
+) ([]entities.ReviewComment, []entities.ReviewComment) {
+	var kept, dropped []entities.ReviewComment
+	for _, comment := range comments {
+		if _, ok := livePaths[normalizeFilePath(comment.FilePath)]; ok {
+			kept = append(kept, comment)
+		} else {
+			dropped = append(dropped, comment)
+		}
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply internal/domain/commands/review_command.go:369]: Addressed in `041874d`. `filterStaleComments` now keeps any comment with `Line <= 0` or an empty `FilePath` — those render as repo-wide annotations via `PostPullRequestComment` and cannot trigger ADO's "file no longer exists" warning, so the staleness drop should never apply to them. Added a regression row `should keep PR-wide comments even when their FilePath looks stale` that pins the contract.

File: internal/domain/commands/review_command.go [hunk 2]
```diff
@@ -288,6 +291,104 @@ func (c *ReviewCommand) postComments(
...
+func summarizeStaleFilePaths(dropped []entities.ReviewComment) string {
+	const maxShown = 8
+	seen := make(map[string]struct{}, len(dropped))
+	paths := make([]string, 0, len(dropped))
+	for _, c := range dropped {
+		if _, ok := seen[c.FilePath]; ok {
+			continue
+		}
+		seen[c.FilePath] = struct{}{}
+		paths = append(paths, c.FilePath)
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply internal/domain/commands/review_command.go]: Addressed in `041874d`. `summarizeStaleFilePaths` now keys its dedup map on the normalised path (the same `normalizeFilePath` rule the membership check uses), so an AI response that mixes `internal/foo.go` and `/internal/foo.go` counts once. The first form encountered is the one printed, so the operator log preserves whichever shape the AI actually emitted. New row `should deduplicate by normalised path so leading-slash variants are not double-counted` pins the contract.

File: internal/domain/commands/export_test.go [hunk 3]
```diff
@@ -8,4 +8,9 @@ package commands
 // repository/provider/registry. The variable indirection keeps the helper
 // itself unexported in production builds (the file is gated on the `unit`
 // build tag).
-var ShouldPostSummary = shouldPostSummary
+var (
+	ShouldPostSummary        = shouldPostSummary
+	FilterStaleComments      = filterStaleComments
+	SummarizeStaleFilePaths  = summarizeStaleFilePaths
+	NormalizeFilePathForTest = normalizeFilePath
+)
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply internal/domain/commands/export_test.go]: Addressed in `041874d`. Rewrote the doc comment to describe the file's purpose (re-exporting unexported helpers for the external `commands_test` package under the `unit` build tag) and added one bullet per re-exported symbol — `ShouldPostSummary`, `FilterStaleComments`, `SummarizeStaleFilePaths`, and `NormalizeFilePathForTest` — so the comment stays accurate as more helpers are added.
